forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/apiclient/ApiClientIdentity.java

```diff
@@ -0,0 +1,156 @@
+    public X509Certificate getTransportCertificate(){
+        X509Certificate[] certificateChain = this.authentication.getCertificateChain();
+        if(certificateChain != null && certificateChain.length >= 1){
+            return certificateChain[0];
+        } else {
+            return null;
+        }
+    }
+
+
+    public boolean isUnregistered(){
+        return authentication.getAuthorities().contains(OBRIRole.UNREGISTERED_TPP);
+    }
+
+
+    public String getUsername() {
+        return authentication.getName();
+    }
+
+    /**
+     * Return the authorities identified by the spring-security-multi-auth.
+     * @return
+     */
+    public Collection<GrantedAuthority> getAuthorities() {
+        return authentication.getAuthorities();
+    }
+
+    /**
+     * Check the transport certificate is issued by a recognised authority. Will throw with errors that can be
+     * directly returned from Client Registration Endpoints.
+     */
+    public void throwIfNotValidCertAuthority() throws OAuth2InvalidClientException {
+        if (this.getAuthorities().contains(OBRIRole.UNKNOWN_CERTIFICATE)) {
+            String helpString = "The transport certificate is not signed by a recognised " +
+                    "authority";
+            X509Certificate certificate = this.getTransportCertificate();
+            if(certificate != null){
+                X500Principal principal = certificate.getIssuerX500Principal();
+                if(principal != null) {
+                    helpString += " '" + principal + "'";
+                }
+            }
+            throw new OAuth2InvalidClientException(helpString);
+        }
+    };
+
+    /**
+     * Check if the TPP was identified as being already onboarded
+     */
+    public void throwIfTppAlreadyOnboarded() throws OAuth2InvalidClientException {
+        log.debug("throwIfTppAlreadyOnboarded() User detail: username: '{}' and authorities: '{}'", this.getUsername(),
+                this.getAuthorities());
+        if (this.getAuthorities().contains(OBRIRole.ROLE_AISP)
+                || this.getAuthorities().contains(OBRIRole.ROLE_PISP)
+                || this.getAuthorities().contains(OBRIRole.ROLE_CBPII)) {
+            String errorMessage = "A software statement has already been registered with this transport certificate.";
+            log.debug("throwIfTppAlreadyOnboarded() {}", errorMessage);
+            throw new OAuth2InvalidClientException(errorMessage);
+        }
+    }
+
+    public void throwIfTppNotOnboarded() throws OAuth2InvalidClientException {
+        log.debug("throwIfTppNotOnboarded() user detail: username '{}', and authorities: '{}'", this.getUsername(),
+                this.getAuthorities());
+        Collection<? extends GrantedAuthority> authorities = this.getAuthorities();
+        if(authorities.contains(OBRIRole.UNREGISTERED_TPP)){
+            String errorString = "Tpp is not onboarded.";
+            log.debug("throwIfTppNotOnboarded() {}", errorString);
```

Comment:
Could be better log.error or log.warn instead of log.debug? because contains the error message.